language_service/src/qsc_utils.rs

```diff
@@ -7,6 +7,13 @@ use qsc::{
     CompileUnit, PackageStore, PackageType, SourceMap, Span,
 };
 
+/// Represents a span of text used by the Language Server API
+#[derive(Debug, PartialEq)]
+pub struct LsSpan {
```

Comment:
I like this new common type, but feel that `qsc_utils.rs` is getting a bit mish-mashy. I would appreciate a new file `types.rs` - especially since I think we're likely to add more common data types soon.

Oh also I prefer just `Span` even though I know overloading names can get a bit frustrating sometimes :) It's just that we're _already_ in the language service module so I'm not a fan of the `Ls` prefix.

Reply:
I actually find calling it `Span` hugely confusing as there is already a commonly referenced type `Span` from the compiler, and just seeing `Span` somewhere makes me think that we are referencing the compiler `Span`. This is why I wanted the span that is specific to the Language Server API to be called something different or prefaced in someway to distinguish it from the other span.

Reply:
Discussed offline. Resolved to create a separate file for defining the protocol surface for the language server that will define the data structures required by the language server protocol. The `Span` type will be defined there.